aten/src/ATen/native/native_functions.yaml

```diff
@@ -3153,6 +3153,23 @@
     CPU: softmax_backward_cpu
     CUDA: softmax_backward_cuda
 
+- func: masked_softmax.int(Tensor self, Tensor mask, int dim, ScalarType? dtype=None) -> Tensor
```

Comment:
nit: can we put both of the non-underscore versions under `python_module: nn` (see other examples in this file).  You'll also have to change the implementation in nn/functional.py to call the updated function, eg. as in https://github.com/pytorch/pytorch/blob/74d6a6106cf7e0629886b23594fb81c0b8e434ab/torch/nn/functional.py#L459.